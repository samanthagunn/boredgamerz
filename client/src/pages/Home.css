.logo-hero {
    position: relative;
    top: 10px;
    height: 80px;
}