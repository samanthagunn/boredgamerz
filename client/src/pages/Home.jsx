import { IonContent, IonPage } from "@ionic/react";
import React from "react";
import Header from "../components/header";
import LandingPage from "../components/landing-page";
import Footer from "../components/footer";

const Home = () => {
  return (
    <IonPage>
      <Header />
      <IonContent className="has-header">
        <LandingPage />
      </IonContent>
      <Footer />
    </IonPage>
  );
};

export default Home;
