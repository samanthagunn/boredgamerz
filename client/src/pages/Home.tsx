import { IonContent, IonPage } from "@ionic/react";
import React from "react";
import Footer from "../components/footer";
import Header from "../components/header";
import LandingPage from "../components/landing-page";
import FAB from "../components/mobile-fab";
import "./Home.css";

const Home = () => {
  return (
    <IonPage>
      <Header />
      <IonContent fullscreen>
        <LandingPage />
      </IonContent>
      <FAB />
      <Footer />
    </IonPage>
  );
};

export default Home;
