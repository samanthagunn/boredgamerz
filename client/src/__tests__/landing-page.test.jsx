import React from "react";
import {
  render,
  fireEvent,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "../pages/Home";
import { useAuth0 } from "@auth0/auth0-react";
import renderer from "react-test-renderer";
import LandingPage from "../components/landing-page";
jest.mock("@auth0/auth0-react");
jest.mock("react-router");

const user = {
  name: "Christian Glassiognon",
  given_name: "Christian",
  picture:
    "https://lh3.googleusercontent.com/a-/AOh14GiG4KuyPH8-ToSqzoBLwPTvb-px4DboWJn_Gujs=s96-c",
};
describe("The Homepage componenet...", () => {
  describe("that is user authenticated...", () => {
    beforeAll(() => {
      useAuth0.mockReturnValue({
        logout: jest.fn(),
        loginWithRedirect: jest.fn(),
        isAuthenticated: true,
        isLoading: false,
        user: user,
      });
    });
    test("will render without crashing", () => {
      const { baseElement } = render(<LandingPage />);
      expect(baseElement).toBeDefined();
    });
    test("will display correct snapshot.", () => {
      const tree = renderer.create(<LandingPage />).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
  describe("that is not user authenticated...", () => {
    beforeAll(() => {
      useAuth0.mockReturnValue({
        logout: jest.fn(),
        loginWithRedirect: jest.fn(),
        isAuthenticated: false,
        isLoading: false,
        user: user,
      });
    });
    test("will render without crashing", () => {
      const { baseElement } = render(<LandingPage />);
      expect(baseElement).toBeDefined();
    });
    test("will match snapshot.", () => {
      const tree = renderer.create(<LandingPage />).toJSON();
      expect(tree).toMatchSnapshot();
  })
    describe("has a Login / Sign Up button...", () => {
      test("that is displayed.", () => {
        const { getByText } = render(<LandingPage />);
        expect(getByText("Login / Sign Up")).toBeTruthy();
      });
      test("that when clicked redirects to authentication.", () => {
        const { getByText } = render(<LandingPage />);
        fireEvent.click(getByText("Login / Sign Up"));
        expect(useAuth0().loginWithRedirect).toHaveBeenCalled();
      });
    });
  });
});
