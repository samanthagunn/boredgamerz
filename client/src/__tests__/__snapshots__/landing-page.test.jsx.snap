// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Homepage componenet... that is not user authenticated... will match snapshot. 1`] = `
<div
  className="container"
>
  <img
    alt="gameplayers"
    className="hero-image"
    src="https://storage.googleapis.com/boredgamerz_assets/gameplayers.png"
  />
  <div
    className="landing-hero-container"
  >
    <ion-card
      color="secondary"
      onClick={[Function]}
    >
      <ion-card-header>
        <h1>
          Easy as Find. Host. Play. 
        </h1>
        <img
          alt="BoredGamerz Logo"
          className="hero-logo"
          src="https://storage.googleapis.com/boredgamerz_assets/BoredGamerzLogo_72-2.png"
          width="40%"
        />
      </ion-card-header>
      <ion-card-content>
        <ion-card-title>
          <strong>
             Step 1
          </strong>
        </ion-card-title>
        <p>
          Find a game by searching in your area
        </p>
      </ion-card-content>
      <ion-card-content>
        <ion-card-title>
          <strong>
            Step 2
          </strong>
        </ion-card-title>
        <p>
          Can't find a game? Host one!
        </p>
      </ion-card-content>
      <ion-card-content>
        <ion-card-title>
          <strong>
            Step 3
          </strong>
        </ion-card-title>
        <p>
          Meet and play the game!
        </p>
      </ion-card-content>
      <ion-button
        onClick={[Function]}
      >
        Login / Sign Up
      </ion-button>
      <br />
      <br />
    </ion-card>
  </div>
</div>
`;

exports[`The Homepage componenet... that is user authenticated... will display correct snapshot. 1`] = `
<div
  className="container"
>
  <img
    alt="gameplayers"
    className="hero-image"
    src="https://storage.googleapis.com/boredgamerz_assets/gameplayers.png"
  />
  <div
    className="landing-hero-container"
  >
    <ion-card
      color="secondary"
      onClick={[Function]}
    >
      <ion-card-header>
        <h1>
          Easy as Find. Host. Play. 
        </h1>
        <img
          alt="BoredGamerz Logo"
          className="hero-logo"
          src="https://storage.googleapis.com/boredgamerz_assets/BoredGamerzLogo_72-2.png"
          width="40%"
        />
      </ion-card-header>
      <ion-card-content>
        <ion-card-title>
          <strong>
             Step 1
          </strong>
        </ion-card-title>
        <p>
          Find a game by searching in your area
        </p>
      </ion-card-content>
      <ion-card-content>
        <ion-card-title>
          <strong>
            Step 2
          </strong>
        </ion-card-title>
        <p>
          Can't find a game? Host one!
        </p>
      </ion-card-content>
      <ion-card-content>
        <ion-card-title>
          <strong>
            Step 3
          </strong>
        </ion-card-title>
        <p>
          Meet and play the game!
        </p>
      </ion-card-content>
      <ion-button
        href="/profile"
        onClick={[Function]}
        router-direction="forward"
      >
        Join a Game 
        Christian
        !
      </ion-button>
      <br />
      <br />
    </ion-card>
  </div>
</div>
`;
